### PR TITLE
refactor(metrics): support streaming metrics level

### LINKED ISSUE #11855: refactor: slim down compute metric size
related #11822

`Histogram` is much more expensive compared to Counter and Gauge. May start from this.

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Closes #11855

Reduce the streaming metrics by 

- disabling all tokio metrics
- folding streaming `Histogram` metrics in fragment-level instead of actor-level

under `steaming_metrics_level = INFO`. Switch to `DEBUG` to get the original behavior.

This PR is based on #11895 and moved some structs to `common` crate in order to reuse them.

Excluding the `state_store` metrics (wich is significantly reduced by #11895), the metrics size will be cut off more than half (8.5MB -> 4.1MB per collecting request).  (Sampled from a customer with 500+ materialized views)

```
-rw-r--r--  1 eric  staff    46M Aug 31 14:10 prom_cn.txt
-rw-r--r--  1 eric  staff   8.5M Aug 31 14:20 prom_cn_without_state_store.txt
-rw-r--r--  1 eric  staff   4.1M Aug 31 14:21 prom_cn_without_state_store_with_this_pr.txt
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/metrics/relabeled_metric.rs
Comment (yuhao-su) [inline src/common/src/metrics/relabeled_metric.rs:35]: Could you add some comments about why we need this?

Comment (fuyufjh) [reply src/common/src/metrics/relabeled_metric.rs:35]: Sure.

**Review by yuhao-su — APPROVED** [review]
LGTM if you have checked the grafana figures locally

**Comment by fuyufjh** [comment]
> LGTM if you have checked the grafana figures locally

Preview:

Screen Shot 2023-08-31 at 19 42 59
Screen Shot 2023-08-31 at 19 43 22
Screen Shot 2023-08-31 at 19 43 44

**Review by zwang28 — APPROVED** [review]
LGTM.
Note that there has already been a `metrics_level` defined in src/common/src/config.rs, which determines whether starts Prometheus for compute node/frontend/compactor. Shall we remove storage_metric_level and streaming_metric_level, extend and use the existent `metrics_level` instead? The downside is we can no longer set metric level for storage and streaming separately.

**Comment by yuhao-su** [comment]
> Shall we remove storage_metric_level and streaming_metric_level, extend and use the existent metrics_level instead? 

Looks like the pr already did this?
https://github.com/risingwavelabs/risingwave/pull/11994/files#diff-c81fa6ff8ebe6cc6822331e0f5ae282664862104a8b9189edce5e9336685690fR570

**Comment by zwang28** [comment]
> > Shall we remove storage_metric_level and streaming_metric_level, extend and use the existent metrics_level instead?
> 
> Looks like the pr already did this? https://github.com/risingwavelabs/risingwave/pull/11994/files#diff-c81fa6ff8ebe6cc6822331e0f5ae282664862104a8b9189edce5e9336685690fR570

No, I mean this one https://github.com/risingwavelabs/risingwave/blob/0132954db2fe3fbdcb93f63f216e6abdc40db353/src/common/src/config.rs#L366

**Comment by fuyufjh** [comment]
> > > Shall we remove storage_metric_level and streaming_metric_level, extend and use the existent metrics_level instead?
> > 
> > 
> > Looks like the pr already did this? https://github.com/risingwavelabs/risingwave/pull/11994/files#diff-c81fa6ff8ebe6cc6822331e0f5ae282664862104a8b9189edce5e9336685690fR570
> 
> No, I mean this one
> 
> https://github.com/risingwavelabs/risingwave/blob/0132954db2fe3fbdcb93f63f216e6abdc40db353/src/common/src/config.rs#L366

+1. Let me merge this PR first 🤣 Didn't notice it was blocked by CI...

File: src/stream/src/task/barrier_manager/managed_state.rs
Comment (fuyufjh) [inline src/stream/src/task/barrier_manager/managed_state.rs:92]: cc. @zwang28 This is a side effect of changing `GLOBAL_STREAMING_METRICS` from `LazyLock` to `OnceLock`. It's difficult to tell whether `GLOBAL_STREAMING_METRICS` is guaranteed to be initialized, so we might have to put a `if let` here to check its availability, which seems a little bit ugly.

**Pull request merged**
